openpype/hosts/maya/plugins/publish/extract_fbx_animation.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+import os
+
+from maya import cmds  # noqa
+import pyblish.api
+
+from openpype.pipeline import publish
+from openpype.hosts.maya.api import fbx
+from openpype.hosts.maya.api.lib import namespaced
+
+
+class ExtractFBXAnimation(publish.Extractor):
+    """Extract Rig in FBX format from Maya.
+
+    This extracts the rig in fbx with the constraints
+    and referenced asset content included.
+    This also optionally extract animated rig in fbx with
+    geometries included.
+
+    """
+    order = pyblish.api.ExtractorOrder
+    label = "Extract Animation (FBX)"
+    hosts = ["maya"]
+    families = ["animation.fbx"]
+
+    def process(self, instance):
+        # Define output path
+        staging_dir = self.staging_dir(instance)
+        filename = "{0}.fbx".format(instance.name)
+        path = os.path.join(staging_dir, filename)
+        path = path.replace("\\", "/")
+
+        fbx_exporter = fbx.FBXExtractor(log=self.log)
+        out_set = instance.data.get("animated_skeleton", [])
+        # Export
+        instance.data["constraints"] = True
+        instance.data["skeletonDefinitions"] = True
+        instance.data["referencedAssetsContent"] = True
+
+        fbx_exporter.set_options_from_instance(instance)
+
+        # Export from the rig's namespace so that the exported
+        # FBX does not include the namespace but preserves the node
+        # names as existing in the rig workfile
+        namespace, relative_out_set = out_set[0].split(":", 1)
+        with namespaced(
+            ":" + namespace,
+            new=False,
+            relative_names=True
+        ) as namespace:
+            fbx_exporter.export(relative_out_set, path)
```

Comment:
We might need to implement these:
```python
def get_namespace(node):
    """Return namespace of given node"""
    return node.rsplit("|", 1)[-1].rsplit(":", 1)[0]


def strip_namespace(node, namespace):
    """Strip given namespace from node path.
    
    The namespace will only be stripped from names
    if it starts with that namespace. If the namespace
    occurs within another namespace it's not removed.
    
    Examples:
        >>> strip_namespace("namespace:node", namespace="namespace:")
        "node"
        >>> strip_namespace("hello:world:node", namespace="hello:world")
        "node"
        >>> strip_namespace("hello:world:node", namespace="hello")
        "world:node"
        >>> strip_namespace("hello:world:node", namespace="world")
        "hello:world:node"
        >>> strip_namespace("ns:group|ns:node", namespace="ns")
        "group|node"
        
    Returns:
        str: Node name without given starting namespace.
    
    """
    
    # Ensure namespace ends with `:`
    if not namespace.endswith(":"):
        namespace = "{}:".format(namespace)
    
    # The long path for a node can also have the namespace
    # in its parents so we need to remove it from each
    return "|".join(
        name[len(namespace):] if name.startswith(namespace) else name
        for name in node.split("|")
    )
```

And then ensure we get and strip correctly for the members we want:
```suggestion
        out_members = instance.data.get("animated_skeleton", [])
        # Export
        instance.data["constraints"] = True
        instance.data["skeletonDefinitions"] = True
        instance.data["referencedAssetsContent"] = True

        fbx_exporter.set_options_from_instance(instance)

        # Export from the rig's namespace so that the exported
        # FBX does not include the namespace but preserves the node
        # names as existing in the rig workfile
        namespace = get_namespace(out_members[0])
        relative_out_members = [
            strip_namespace(node, namespace) for node in out_members
        ]
        with namespaced(
            ":" + namespace,
            new=False,
            relative_names=True
        ) as namespace:
            fbx_exporter.export(relative_out_members, path)
```